scipy/ndimage/src/ni_morphology.c

```diff
@@ -623,6 +623,10 @@ int NI_DistanceTransformBruteForce(PyArrayObject* input, int metric,
             temp->index = jj;
             temp->coordinates = malloc(PyArray_NDIM(input) *
                                        sizeof(npy_intp));
+            if (!temp->coordinates) {
+                PyErr_NoMemory();
```

Comment:
`temp` needs to be freed here.

Reply:
`temp` is assigned to `border_elements` above which is singly linked list. So, it was fine as it was before.

Reply:
Ok, indeed, it shouldn't be freed here.

Reply:
I reverted the old commit.